### PR TITLE
Feature/113 Paginator enhancement

### PR DESCRIPTION
**Pull request by moetazaneta**

### CODE REVIEW

File: src/components/Paginator/Paginator.tsx [hunk 1]
```diff
@@ -19,12 +20,21 @@ interface Props {
    * and how many items would be visible from sides of middle.
    */
   readonly edgeCount?: number;
+
+  /** Whether paginator have `next` and `previous` buttons. */
+  readonly isNavigationHidden?: boolean;
```
Comment (NightOfNights) [inline src/components/Paginator/Paginator.tsx]:
```suggestion

  /** Whether paginator `next` and `previous` buttons should be hidden or not. */
  readonly isNavigationHidden?: boolean;
```

File: src/components/Paginator/Paginator.tsx [hunk 2]
```diff
@@ -47,14 +57,29 @@ const PaginatorComponent: VFC<Props> = ({ count, currentPage, onChange, edgeCoun
   const [leftEdgePages, middlePages, rightEdgePages] = getButtons();
 
   const isMiddlePagesVisible = middlePages.length > 0;
-  const isRightEdgePagesVisible = rightEdgePages.length > 0;
+  const isRightEdgePagesVisible = rightEdgePages.length > 0 && currentPage > edgeCount + 2;
```
Comment (NightOfNights) [inline src/components/Paginator/Paginator.tsx]: Why `+ 2`?

File: src/components/Paginator/Paginator.tsx [hunk 3]
```diff
@@ -47,14 +57,29 @@ const PaginatorComponent: VFC<Props> = ({ count, currentPage, onChange, edgeCoun
...
 
   /**
    * Handle page click.
    * @param page Page number.
    */
   const handlePageClick = useCallback((page: number) => onChange(page), [onChange]);
 
+  /**
+   * Handle next page click.
```
Comment (NightOfNights) [inline src/components/Paginator/Paginator.tsx]: Function does what? Function _handles_ (not handle) next page click

File: src/components/Paginator/PaginatorNavigationButton.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import { IconButton } from '@chakra-ui/react';
+import { Icon } from '../Icon/Icon';
+
+interface Props {
+
+  /** Whether button is disabled. */
+  readonly isDisabled?: boolean;
+
+  /** Click handler. */
+  readonly onClick?: MouseEventHandler<HTMLButtonElement>;
```
Comment (NightOfNights) [inline src/components/Paginator/PaginatorNavigationButton.tsx]: prefer `() => void` like signatures for consistency

File: src/features/VisualNovel/pages/VisualNovelPage/VisualNovelPage.module.css [hunk 5]
```diff
@@ -1,10 +1,6 @@
-/*
- * TODO (Sotnichenko): Adopt scrollbar library
- * Nearly all sub routes make page scrollable,
- * so to prevent screen size flicking we add scrollbar for all pages.
- */
 html {
-  overflow-y: scroll;
+  overflow-y: auto;
+  scrollbar-gutter: stable;
```
Comment (NightOfNights) [inline src/features/VisualNovel/pages/VisualNovelPage/VisualNovelPage.module.css]: Shouldn't it be global-scope style?

File: src/components/Paginator/Paginator.tsx [hunk 6]
```diff
@@ -19,42 +20,77 @@ interface Props {
...
-    if (currentPage >= count - edgeCount - 1) {
-      const rightEdgePages = Array.from({ length: count - currentPage + edgeCount + 1 }).map((_, i) => count - i)
+    if (currentPage >= count - edgeCount - inputMinGap) {
+      const rightEdgePages = Array.from({ length: count - currentPage + edgeCount + offset }).map((_, i) => count - i)
         .reverse();
       return [leftEdgePages, [], rightEdgePages];
     }
 
-    const middlePages = Array.from({ length: 1 + edgeCount * 2 }).map((_, i) => i + (currentPage - edgeCount));
+    const middlePages = Array.from({ length: edgeCount * 2 + offset }).map((_, i) => i + (currentPage - edgeCount));
```
Comment (NightOfNights) [inline src/components/Paginator/Paginator.tsx]: Why `* 2` ? 🙃

File: src/components/Paginator/Paginator.tsx [hunk 7]
```diff
@@ -19,42 +20,77 @@ interface Props {
...
 
   const [leftEdgePages, middlePages, rightEdgePages] = getButtons();
 
   const isMiddlePagesVisible = middlePages.length > 0;
-  const isRightEdgePagesVisible = rightEdgePages.length > 0;
+  const isRightEdgePagesVisible = rightEdgePages.length > 0 && currentPage > edgeCount + offset + inputMinGap;
 
   /**
-   * Handle page click.
+   * Handles page click.
```
Comment (NightOfNights) [inline src/components/Paginator/Paginator.tsx:73]: What about docs for other functions? 🙃

File: src/components/Paginator/Paginator.tsx [hunk 8]
```diff
@@ -19,42 +20,77 @@ interface Props {
...
+  readonly isNavigationHidden?: boolean;
+
+  /**
+   * Pagination offset from 0.
+   * Defaults to 1, so pagination is user friendly.
+   */
+  readonly offset?: number;
+
+  /** Minimum gap between two edge buttons groups for input button(ellipsis) to appear. */
+  readonly inputMinGap?: number;
```
Comment (NightOfNights) [inline src/components/Paginator/Paginator.tsx]:
```suggestion
  /** Minimum gap between edge buttons for page number input to appear. */
  readonly inputMinGap?: number;
```

File: src/components/Paginator/Paginator.tsx (hunk 6 above)
Comment (moetazaneta) [reply src/components/Paginator/Paginator.tsx]: Middle group has only two sides

**Pull request merged**